### DISCUSSION TITLE
Will blazor-binding be supported in the future?

### DISCUSSION THREAD

**Discussion by CeSun** [AvaloniaUI/Avalonia#9211] [Ideas]
Xamarin-blazor-binding: https://learn.microsoft.com/en-us/mobile-blazor-bindings/
BlazorBindings.Maui: https://github.com/Dreamescaper/BlazorBindings.Maui

Razor is more flexible than XAML，Hopefully this feature will be added in the future.

**Comment by kekekeks** [comment]
Not currently planned, but there is nothing that prevents such bindings from existing. 
The closest thing we have right now is [FuncUI](https://funcui.avaloniaui.net/guides/basic-template-mvu) which works in a similar way but uses F# type providers.

**Reply by MarcoMedrano** [reply]
Blazor binding is way way simpler than mvvm, reactiveUI, or whatever state based tech like react.js redux, mobx, angular etc. etc.

You guys should consider using it in Avalonia.

**Comment by robloo** [comment]
There is actually already an issue for this: https://github.com/AvaloniaUI/Avalonia/issues/6267#issue-946687251

I would like to see the Razor syntax used in Avalonia in the years to come as well. I guess this is far away though when there is time to actually re-think how things were done instead of trying to complete missing functionality.

Also see: https://github.com/AvaloniaUI/Avalonia/issues/2502. Specifically the discussion starting here: https://github.com/AvaloniaUI/Avalonia/issues/2502#issuecomment-631845909

**Comment by timahrentlov** [comment]
I believe that Mobile Blazor Bindings - [like the ones for MAUI](https://learn.microsoft.com/en-us/mobile-blazor-bindings/) - would benefit Avalonia in general.  

To be able to use Razor with Avalonia Controls, to mix and match with HTML/CSS or to be deployed as the UI in an [.NET MAUI Blazor app](https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/tutorials/maui?view=aspnetcore-6.0) would catapult Avalonia to the main stage.

**Reply by robloo** [reply]
Razor/Blazor syntax would be nice.

Integration with HTML seems opposite of the project goals. Most of us are here because HTML/Javascript apps are not ideal for our scenarios. Personally, I think web apps never should have become dominant. The tech is simply inferior.

**Reply by timahrentlov** [reply]
I agree, but the value proposition of being able to mix and match both would benefit Avalonia as such. More flies caught. Really just want Avalonia to become "mainstream". Depending on how it is done, of course.

**Comment by MarcoMedrano** [comment]
> Razor/Blazor syntax would be nice.
> 
> Integration with HTML seems opposite of the project goals. Most of us are here because HTML/Javascript apps are not ideal for our scenarios. Personally, I think web apps never should have become dominant. The tech is simply inferior.

HTML is inferior, totally agree, but the proposal here is to use blazor bindings (no HTML tech) , which basically redices binding and almost eliminates explicit binding, making development process way more natural and faster.

**Reply by robloo** [reply]
I was replying to the quote:

> to mix and match with HTML/CSS

Which very much is implying usage of HTML/CSS. I don't mind if a simpler binding syntax is used.

**Reply by MarcoMedrano** [reply]
No,  we could use Xaml instead, that is the proposal.
